the-collab-lab/tcl-25-smart-shopping-list
# Add an estimated number of days until the next purchase date

## Description

After the purchase is recorded, this PR calculates the estimated number of days until the next purchase date and records it in the database. 



## Related Issue
#11 


## Acceptance Criteria




## Type of Changes



|     | Type                       |
| --- | -------------------------- |
|    | :bug: Bug fix              |
|  ✓ | :sparkles: New feature     |
|    | :hammer: Refactoring       |
|    | :100: Add tests            |
|    | :link: Update dependencies |
|    | :scroll: Docs              |


## Testing Steps / QA Criteria



- From your terminal, pull down this branch with `git pull origin ci-io-estimated-purchase-date` and check that branch out with `git checkout ci-io-estimated-purchase-date`.
-  Add an item using the `add view` form
-  Navigate back to the  `list view`
-  Check the newly added item
-   Check the database [here](https://console.firebase.google.com/project/tcl-25-shopping-list/firestore/data~2Flists~2F6FzzLNEJ9kXz1Jpeacqn),  the newly added item will have a new field `numberOfPurchases` and the `daysLeftBeforeTheNextPurchase` field will be updated.
